backend/routes/endSession.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const router = express.Router();
+const admin = require('firebase-admin');
+// const advanceQueue = require('./advanceQueue'); // Import advanceQueue function
+
+// End Session Endpoint
+router.post('/endSession', async (req, res) => {
+    try {
+        // Extract location and user UID from the request body
+        const { location, uid } = req.body
+        if (!location || !uid || uid.toLowerCase().startsWith("empty")) {
+            return res.status(400).json({ message: 'Valid location and UID are required.' });
+        }
+
+        // Reference to activePlayers
+        const activePlayersRef = admin.firestore()
+            .collection('locations')
+            .doc(location)
+            .collection('activePlayers');
+
+        // Check if the player exists in activePlayers (if found, location is implicitly valid)
+        const activePlayerSnapshot = await activePlayersRef
+            .where('firebaseUID', '==', uid)
+            .limit(1)
+            .get();
+
+        if (activePlayerSnapshot.empty) {
+            return res.status(404).json({ message: 'Player or location not found.' });
+        }
+    
+        // Delete the player from active players
+        const activePlayerDoc = activePlayerSnapshot.docs[0];
```

Comment:
Similar to the /resetCourts and /addUnknowns endpoint, the definition of removing a player is not deleting the entry, but updating the attributes. In particular, nickname / firebaseUID = "Empty{X}", x = court #. playerWaiting = False and remaining 3 leave same... Those endpoints provide examples of the .update method with merge = True.